### ISSUE TITLE
Supabase session being reset on every page load

### ISSUE THREAD

**Issue by bbarr** [supabase/auth-js#995]
https://github.com/supabase/auth-js/blob/b045965298863ea5a085b96b2e21d5dfd31175ec/src/GoTrueClient.ts#L310

I notice that because `detectSessionInUrl` is always `true`, the auth code is looking for the session in the URL and when it isn't there (every load other than magic link, in my application), it is treating it as an error and resetting.

This wasn't happening the other day, and I don't think I have changed my Supabase version, so it must be a problem with my implementation somehow.. and yet, when I walk through the trace, it clearly gets to this point in the auth library every load, and of course, there is no session info still in the URL, and it kills the session. Can someone better help me understand how `detectSessionInUrl` is supposed to work?

Thanks!

**Comment by Eirmas** [comment]
I notice this issue too, and I cannot make sense of their logic.
Their `initialize` function starts by checking the URL for a session, and if its not there, they remove the data in localstorage, and triggers an "SIGNED_OUT" event. Why?

This basically means every time you refresh the page without a session in the URL, you will be logged out.

**Comment by bbarr** [comment]
SOLVED. I think I have found a bigger issue. The following code has an early exit that needs to fire in order to avoid losing the session. 

https://github.com/supabase/auth-js/blob/b045965298863ea5a085b96b2e21d5dfd31175ec/src/GoTrueClient.ts#L315-L329

But `_getSessionFromUrl()` is returning the right AuthImplicitGrantRedirectError, but without the `details` required to hit the early exit, seen here:

https://github.com/supabase/auth-js/blob/b045965298863ea5a085b96b2e21d5dfd31175ec/src/GoTrueClient.ts#L1443-L1445

Note the lack of a second argument to the error constructor, which is where the `details` would be. 

This commit just happened 11 hours ago: https://github.com/supabase/auth-js/commit/089c687b24da8d48de8302ea201c04dcf5e928a8 which features the new Error call without `details`.

I can confirm that this is resolved by dropping the SDK version to 2.47.5, which is the one before the last commit.

**Comment by harrisrobin** [comment]
After hours of debugging, this is why our app keeps logging users out after a few page refreshes! Insane that this made it through as a patch without us changing any supabase versions across our app.

I was able to fix our app by adding the following overrides and reverting back.

```
  "overrides": {
    "@supabase/auth-js": "2.66.1"
  },
  ```

**Comment by ufukcam** [comment]
I'm using the @nuxtjs/supabase package in my Nuxt app, and I’ve noticed that with detectSessionInUrl set to true, the auth code keeps checking for a session in the URL. Since my app doesn’t rely on magic links often, there’s no session info in the URL, and it ends up resetting the session on every load.

This just started happening recently—I haven’t changed my Supabase or Nuxt package versions. When I trace through the code, it always reaches this point in the auth library and invalidates the session because it can't find the session in the URL.

Can someone clarify how detectSessionInUrl is supposed to behave with the Nuxt package, or if I might have misconfigured something?

Thanks!

**Comment by Shooteger** [comment]
We have a big Nuxt 3 + Supabase web app.
We encountered a similar issue to the one described by @harrisrobin. After updating packages locally and pushing to the production-testing branch, our application started logging users out on every page route or navigation due to a "Session missing" error.

In our local development environment, the issue was less consistent but still occurred randomly, often after some navigation. This made debugging particularly challenging.

After extensive debugging, we traced the issue to the latest version of @supabase/auth-js. Reverting to a previous version resolved the problem. Specifically:

The issue does not occur when using @supabase/auth-js version 2.66.1.
Using any version beyond 2.66.1 caused the "Session missing" error.

Relevant packages in our package.json:
```
"@nuxtjs/supabase": "^1.4.3",
"nuxt": "3.14.1592",
```

The solution from harris is a good temporary fix. You can add overrides to our package.json to force @supabase/auth-js to use version 2.66.1:

> After hours of debugging, this is why our app keeps logging users out after a few page refreshes! Insane that this made it through as a patch without us changing any supabase versions across our app.
> 
> I was able to fix our app by adding the following overrides and reverting back.
> 
> ```
>   "overrides": {
>     "@supabase/auth-js": "2.66.1"
>   },
> ```

**Comment by hf** [comment]
Reverting the change ASAP. Sorry everyone!

**Comment by probook-lewis** [comment]
Thanks, can someone please help to bump the auth-js package to 2.67.1 in supabase-js?

**Comment by onurusluca** [comment]
Yep having this now. Been going crazy for 2 days. Couldn't find any solutions.

**Comment by czarsbenemoriendi** [comment]
I had this issue as well, upon debugging, then downgrading auth-js package the issue was partly solved. On chromium based browsers (tested in Arc and Chrome) I don't have issues anymore, but Safari makes this issue more troublesome.

After investigation I've got an trace that in Safari for below custom storage values aren't kept they are just disappear. After commented below lines and just simply assigning `this.storage = window.localStorage"` the issue is disappearing.

https://github.com/supabase/auth-js/blob/9751b8029b4235a63dcb525e7ce7cc942c85daf5/src/GoTrueClient.ts#L236-L250

I've passed `debugging flag to true` in `nuxt.config` cause, I'm using Nuxt 3 with wrapper for Supabase `@nuxt/supabase`


In comparison Chrome console
![chrome-sb](https://github.com/user-attachments/assets/824f663a-256b-4696-b30b-4dff84ffb53e)

and Safari
![safari-sb](https://github.com/user-attachments/assets/74956b71-90a7-40a3-8893-d52b55a6e512)

In Safari code verifier it's empty, this value should be taken from custom storage because storage is empty or just isn't exist that request return error in Safari. Initialisation login flow should kept tokens and necessary values but they just disappearing
